### PR TITLE
feat: add map picker for land registration

### LINKED ISSUE #240: Latitude and longitude should be picked from map on land registraion
For register land on frontend, users should be able to add longitude and latitude with a map picker.

### PR DESCRIPTION
**Pull request by MarcusDavidG**
- Add `MapPicker` component for visual coordinate selection  
- Integrate map picker into `RegisterLandModal`  
- Replace manual coordinate input with interactive map  
- Add visual marker for selected location  
- Display selected coordinates below map  

### Title  
**Add map picker for land registration coordinates**  

#### Changes Made  
- Added new `MapPicker` component for visual coordinate selection  
- Integrated interactive map into `RegisterLandModal`  
- Replaced manual coordinate input fields with map interface  
- Added visual marker to show selected location  
- Added coordinate display below map  

#### Features  
- Users can now click on a map to select land coordinates  
- Real-time display of selected latitude and longitude  
- Visual marker shows the exact selected location  
- Map is fully interactive with zoom and pan controls  
- Maintains all existing form validation and submission logic  

#### Technical Details  
- Uses `react-leaflet` for map implementation  
- Dynamically loaded to avoid SSR issues  
- Properly handles marker icon assets  
- Maintains TypeScript type safety  
- Preserves existing form functionality  

### Testing  

To test the changes:  
1. Navigate to the land registration form.  
2. Click on the map to select a location.  
3. Verify the coordinates update.  
4. Complete other form fields.  
5. Submit the form with the selected location.  

---

## Related Issues  
Closes #240  

---

## Type of Change  

- [x] 🐛 Bug fix or ⚙️ Enhancement  
- [x] ✨ New feature or chore (change with no new features or fixes)  
- [x] 📚 Documentation update  

---

## Checklist (select as many as applicable)  

- [x] The code follows the style guidelines of this project.  
- [x] All new and existing tests pass.  
- [x] This pull request is ready to be merged and reviewed.

### CODE REVIEW

**Review by Solomonsolomonsolomon — CHANGES_REQUESTED** [review]
You installed a package,yet the package.json was not modified, please fix

**Review by MarcusDavidG — COMMENTED** [review]
Successfully updated the package.json by committing the package.json and package-lock.json files that include the Leaflet dependencies:
leaflet: ^1.9.4
react-leaflet: ^5.0.0
@types/leaflet: ^1.9.15

**Comment by fishonamos** [comment]
@Solomonsolomonsolomon

File: app/client/package-lock.json
Comment (Solomonsolomonsolomon) [inline app/client/package-lock.json:1]: we dont use package-lock as we use pnpm not npm in app/client.. revert

**Review by Solomonsolomonsolomon — COMMENTED** [review]
App fails, it crashes when you open the register land modal. please fix ASAP

**Comment by MarcusDavidG** [comment]
Done! All issues fix and app smoothly running.

**Pull request closed without merging**